assets/app/views/builds.html

```diff
@@ -8,21 +8,86 @@
+          <ul>
+            <li ng-repeat="trigger in buildConfig.triggers">
+              <span ng-switch="trigger.type">
+                <span ng-switch-when="github">
+                  GitHub webhook URL: <input class="small" readonly="readonly" value="{{buildConfigName | webhookURL : trigger.type : trigger.github.secret : project.metadata.name}}" select-on-focus />
+                  <a href="http://docs.openshift.org/latest/using_openshift/builds.html#webhook-triggers" target="_blank"><i class="pficon pficon-help"></i></a>
+                </span>
+                <span ng-switch-when="generic">
+                  Generic webhook URL: <input class="small" readonly="readonly" value="{{buildConfigName | webhookURL : trigger.type : trigger.generic.secret : project.metadata.name}}" select-on-focus />
+                  <a href="http://docs.openshift.org/latest/using_openshift/builds.html#webhook-triggers" target="_blank"><i class="pficon pficon-help"></i></a>
+                </span>
+                <span ng-switch-when="imageChange">
+                  Image change from image {{trigger.imageChange.image}}:{{trigger.imageChange.tag}}
+                </span>
+                <span ng-switch-default>{{trigger.type}}</span>
+              </span>
+            </li>
+          </ul>
+        </div>
+        <div style="margin-bottom: 10px;" ng-repeat="build in buildsByBuildConfig[buildConfigName]">
+          <h3>{{build.metadata.name}}</h3>
+          <div>Created: <relative-timestamp timestamp="build.metadata.creationTimestamp"></relative-timestamp></div>            
+          <div>Status: {{build.status}}</div>
+          <div ng-if="buildConfig.parameters.strategy.type != build.parameters.strategy.type">
+            <div>Build strategy: {{build.parameters.strategy.type}}</div>
+            <div ng-if="build.parameters.strategy.type == 'STI'">
+              Builder image: {{build.parameters.strategy.stiStrategy.builderImage}}
+            </div>
+          </div>
+          <div ng-if="build.parameters.source">
+            <div ng-if="buildConfig.parameters.source.type != build.parameters.source.type">
+              <div ng-if="build.parameters.source.type == 'Git'">Source repo: {{build.parameters.source.git.uri}}</div>
+            </div>
+            <!-- TODO add git ref details -->
+          </div>
+          <div ng-if="buildConfig.parameters.output.to.kind != build.parameters.output.to.kind && buildConfig.parameters.output.to.name != build.parameters.output.to.name">
+            <div ng-switch="build.parameters.output.to.kind">
+              <div ng-switch-when="ImageRepository">
+                Output image: {{buildConfig.parameters.output.to.name}}
+              </div>
+            </div>
+          </div>
+        </div>
+
+        <!-- render any builds whose build configs no longer exist -->
+        <div style="margin-bottom: 10px;" ng-repeat="build in builds" ng-if="!buildConfigs[build.metadata.labels.buildconfig]">
```

Comment:
Maybe loop over buildsByBuildConfig and combine with `ng-if="!buildConfigs[buildConfig]"` so we can group related builds, even if their buildConfig no longer exists